File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        size = len(nums) * 2
        concated_nums = [0] * size
        concated_nums[0:len(nums)] = nums
        concated_nums[len(nums):] = nums

        return concated_nums

File: Data Structures & Algorithms/combination-target-sum/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums: number[], target: number): number[][] {
        const res: number[][] = []
        const subset: number[] = []

        const dfs = (i: number, remaining: number) => {
            if (remaining < 0 || i >= nums.length) {
                return
            }

            if (remaining === 0) {
                res.push(Array.from(subset))
                return
            }

            subset.push(nums[i])
            dfs(i, remaining - nums[i])

            subset.pop()
            dfs(i + 1, remaining)
        }

        dfs(0, target)

        return res
    }
}
